backend/src/packages/achievements/achievement.service.ts

```diff
@@ -31,6 +35,46 @@ class AchievementService implements IService {
     };
   }
 
+  public async findOwnProgress(
+    userId: number,
+  ): Promise<AchievementWithProgressResponseDto[]> {
+    const achievements = await this.findAll();
+
+    const userAchievements =
+      await this.achievementRepository.findAllUserAchievements(userId);
+
+    const counts = await this.achievementRepository.getCountsInReferenceTables(
+      userId,
+    );
+
+    return achievements.items.map((achievement) => {
+      const isAchieved = userAchievements.find(
+        ({ achievementId }) => achievement.id === achievementId,
+      );
+
+      if (isAchieved) {
+        return {
+          ...achievement,
+          progress: PercentageProgress.MAX,
+        };
+      }
+
+      const referenceNameAndCount = counts.find(
+        (count) => Object.keys(count)[0] === achievement.referenceTable,
+      );
+      const count = referenceNameAndCount
+        ? referenceNameAndCount[achievement.referenceTable]
+        : 0;
```

Comment:
After changes with grouping in repository
```suggestion
      const countByAchievementReference = referenceTableToCount[achievement.referenceTable] ?? PercentageProgress.MIN;
```